client/ayon_resolve/plugins/load/load_editorial_package.py

```diff
@@ -32,21 +36,126 @@ def load(self, context, name, namespace, data):
 
         project = lib.get_current_project()
         media_pool = project.GetMediaPool()
+        folder_path = context["folder"]["path"]
 
         # create versioned bin for editorial package
         version_name = context["version"]["name"]
-        bin_name = f"{name}_{version_name}"
-        lib.create_bin(bin_name)
-
+        loaded_bin = lib.create_bin(f"{folder_path}/{name}/{version_name}")
+        loaded_timeline_name = f"{name}_{version_name}_timeline"
         import_options = {
-            "timelineName": "Editorial Package Timeline",
+            "timelineName": loaded_timeline_name,
             "importSourceClips": True,
             "sourceClipsPath": search_folder_path.as_posix(),
         }
 
+        # import timeline from otio file
         timeline = media_pool.ImportTimelineFromFile(files, import_options)
+
+        # get timeline media pool item for metadata update
+        timeline_media_pool_item = None
+        for item in lib.iter_all_media_pool_clips(root=loaded_bin):
+            item_name = item.GetName()
+            if item_name != loaded_timeline_name:
+                continue
+            timeline_media_pool_item = item
+            break
+
+        # Update the metadata
+        if timeline_media_pool_item:
+            clip_data = self._get_container_data(
+                context, data)
+
+            timeline_media_pool_item.SetMetadata(
+                lib.pype_tag_name, json.dumps(clip_data)
+            )
+
+            # set clip color based on random choice
+            clip_color = self.get_clip_color()
+            print("Clip color: ", clip_color)
+            timeline_media_pool_item.SetClipColor(clip_color)
+
+        # TODO: there are two ways to import timeline resources (representation
+        #   and resources folder) but Resolve seems to ignore any of this
+        #   since it is importing sources automatically. But we might need
+        #   to at least set some metadata to those loaded media pool items
         print("Timeline imported: ", timeline)
```

Comment:
Should this be `self.log.info` calls?

Reply:
yeah I wish so, but it does not log anything so print is more safe.

Reply:
Doesn't that just mean that maybe we need to configure a custom logging handler for Resolve that maybe ends up just doing the print calls?